### PR TITLE
Improve job search

### PR DESCRIPTION
**Pull request by Mari-Hayashi**
Modified job search API (api/job/read) so that it removes jobs with the status "closed" and jobs that have already been applied by logged in student from the search result.

### CODE REVIEW

File: src/server/src/modules/job.ts [hunk 1]
```diff
@@ -132,6 +135,13 @@ export const getJobs = (
         let year = modStart.getFullYear();
         modStart = month + '/' + date + '/' + year;
     }
+
+    const jobApplications = await getJobApplications(studentId);
+
+    if (!jobApplications) return undefined;
```
Comment (johan1505) [inline src/server/src/modules/job.ts:142]: Why you return undefined if no job applications from the student are found?

Comment (Mari-Hayashi) [reply src/server/src/modules/job.ts:142]: If no job applications are found, the jobApplications array is empty without being undefined. The array is undefined only when the studentId passed in as an argument does not exist, which should never happen in this context.

Comment (johan1505) [reply src/server/src/modules/job.ts:142]: ohhhhhhh, can you just add a comment that explains that? I think it'd be good to clear it up.

Comment (johan1505) [reply src/server/src/modules/job.ts:142]: After that feel free merge the pr :D

**Pull request merged**